File: Data Structures & Algorithms/jump-game/submission-3.java
class Solution {
    public boolean canJump(int[] nums) {
        int maxReachable = 0;
        for (int i = 0; i < nums.length; i++) {
            if (maxReachable < i) {
                return false;
            }

            maxReachable = Math.max(maxReachable, i + nums[i]);
        }
        return true;
    }
}

File: Data Structures & Algorithms/binary-search/submission-3.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums: number[], target: number): number {

        let loIdx = 0;
        let highIdx = nums.length - 1;

        while (loIdx <= highIdx) {
            let midIdx = Math.floor((highIdx + loIdx)/2);
            let currentNumber = nums[midIdx];

            if (currentNumber === target) {
                return midIdx;
            }
            if (currentNumber > target) {
                highIdx = midIdx - 1;
            } else {
                loIdx = midIdx + 1;
            }
        }


        return -1;
    }
}
